redash/handlers/query_results.py

```diff
@@ -276,6 +277,16 @@ def get(self, query_id=None, query_result_id=None, filetype='json'):
             if should_cache:
                 response.headers.add_header('Cache-Control', 'private,max-age=%d' % ONE_YEAR)
 
+            str_date = query_result.retrieved_at.strftime("%Y_%m_%d")
+
+            if query is not None:
+                filename = query.name.replace(" ", "_") + "_" + str_date
+            else:
+                filename = str(query_result.id) + "_" + str_date
+            filename += "." + filetype
```

Comment:
On second thought, I think that on the backend it will be safer to just use the query id (when available) instead of the name to avoid any unicode issues 😔 (for example, at the moment, line 288 will probably throw an error if the query name has unicode characters)

I would also use `.format` here. Something along the lines of:

```python
if query is not None:
    prefix = query.id
else:
    prefix = query_result.id

filename = "{}_{}.{}".format(prefix, str_date, filetype)
```

What do you think?

Reply:
Oh silly me! I literally had `format` method used right below that code (when using `add_header`). I've made some changes according to it.